src/battle_main.c

```diff
@@ -299,27 +299,220 @@ const struct OamData gOamData_BattleSpritePlayerSide =
+        TERA_SHARD(GHOST)
+    },
+    [TYPE_STEEL] =
+    {
+        .name = _("Steel"),
+        GENERIC(a STEEL),
+        .palette = 13,
+        MOVES(CORKSCREW_CRASH, STEELSPIKE)
+        ITEMS(STEEL, METAL_COAT, BABIRI, STEEL)
+        LEGENDARY_ITEMS(IRON, STEEL)
+        TERA_SHARD(STEEL)
+    },
+    [TYPE_MYSTERY] =
+    {
+        .name = _("???"),
+        GENERIC(a ???),
+        .palette = 15,
+    },
+    [TYPE_FIRE] =
+    {
+        .name = _("Fire"),
+        GENERIC(a FIRE),
+        .palette = 13,
+        MOVES(INFERNO_OVERDRIVE, FLARE)
+        ITEMS(FIRE, CHARCOAL, OCCA, FIR)
+        LEGENDARY_ITEMS(FLAME, FIRE)
+        TERA_SHARD(FIRE)
+    },
+    [TYPE_WATER] =
+    {
+        .name = _("Water"),
+        GENERIC(a WATER),
+        .palette = 14,
+        MOVES(HYDRO_VORTEX, GEYSER)
+        ITEMS(WATER, MYSTIC_WATER, PASSHO, WATER)
+        LEGENDARY_ITEMS(SPLASH, WATER)
+        TERA_SHARD(WATER)
+    },
+    [TYPE_GRASS] =
+    {
+        .name = _("Grass"),
+        GENERIC(a GRASS),
+        .palette = 15,
+        MOVES(BLOOM_DOOM, OVERGROWTH)
+        ITEMS(GRASS, MIRACLE_SEED, RINDO, GRASS)
+        LEGENDARY_ITEMS(MEADOW, GRASS)
+        TERA_SHARD(GRASS)
+    },
+    [TYPE_ELECTRIC] =
+    {
+        .name = _("Electr"),
+        GENERIC(an ELECTRIC),
+        .palette = 13,
+        MOVES(GIGAVOLT_HAVOC, LIGHTNING)
+        ITEMS(ELECTRIC, MAGNET, WACAN, ELECTR)
+        LEGENDARY_ITEMS(ZAP, ELECTRIC)
+        TERA_SHARD(ELECTRIC)
+    },
+    [TYPE_PSYCHIC] =
+    {
+        .name = _("Psychc"),
+        GENERIC(a PSYCHIC),
+        .palette = 14,
+        MOVES(SHATTERED_PSYCHE, MINDSTORM)
+        ITEMS(PSYCHIC, TWISTED_SPOON, PAYAPA, PSYCH)
+        LEGENDARY_ITEMS(MIND, PSYCHIC)
+        TERA_SHARD(PSYCHIC)
+    },
+    [TYPE_ICE] =
+    {
+        .name = _("Ice"),
+        GENERIC(a ICE),
+        .palette = 14,
+        MOVES(SUBZERO_SLAMMER, HAILSTORM)
+        ITEMS(ICE, NEVER_MELT_ICE, YACHE, IC)
+        LEGENDARY_ITEMS(ICICLE, ICE)
+        TERA_SHARD(ICE)
+    },
+    [TYPE_DRAGON] =
+    {
+        .name = _("Dragon"),
+        GENERIC(a DRAGON),
+        .palette = 15,
+        MOVES(DEVASTATING_DRAKE, WYRMWIND)
+        ITEMS(DRAGON, DRAGON_FANG, HABAN, DRAGON)
+        LEGENDARY_ITEMS(DRACO, DRAGON)
+        TERA_SHARD(DRAGON)
+    },
+    [TYPE_DARK] =
+    {
+        .name = _("Dark"),
+        GENERIC(a DARK),
+        .palette = 13,
+        MOVES(BLACK_HOLE_ECLIPSE, DARKNESS)
+        ITEMS(DARK, BLACK_GLASSES, COLBUR, DARKIN)
+        LEGENDARY_ITEMS(DREAD, DARK)
+        TERA_SHARD(DARK)
+    },
+    [TYPE_FAIRY] =
+    {
+        .name = _("Fairy"),
+        GENERIC(a FAIRY),
+        .palette = 14,
+        MOVES(TWINKLE_TACKLE, STARFALL)
+        ITEMS(FAIRY, FAIRY_FEATHER, ROSELI, FAIR)
+        LEGENDARY_ITEMS(PIXIE, FAIRY)
+        TERA_SHARD(FAIRY)
+    },
+    /*
+    [TYPE_STELLAR] =
+    {
+        .name = _("Stellar"),
+        TERA_SHARD(STELLAR),
+    },
+    */
 };
```

Comment:
Based on precedent, I think I'm in the minority here but this very much seems like a case of macro overuse. Only two of the *twelve* fields of the struct are actually visible here. And MOVES, ITEMS, LEGENDARY_ITEMS do not make It clear what they're actually representing.